improver/utilities/temporal.py

```diff
@@ -144,9 +145,10 @@ def iris_time_to_datetime(
     return datetime_list
 
 
-def datetime_to_iris_time(dt_in: datetime) -> float:
+def datetime_to_iris_time(dt_in: Union[datetime, cftime.DatetimeGregorian]) -> float:
```

Comment:
The type definition is not quite right: the return type annotation should be `np.int64`. Alternatively, the return type could possibly be changed to a Python `int`.